### PR TITLE
Dynamic CORS variables

### PR DESCRIPTION
**Pull request by mark-juwai**
## Summary of changes

- Make the CORS variables configurable.

## How to Test

- Run `$ cd tests && vagrant up`
- Run `$ vagrant ssh -c 'cat /etc/nginx/conf.d/test.conf'`, you can see CORS related config items `Access-Control-*` in the nginx config file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,9 +57,25 @@ Retention of the logrotate.
...
 Default is 30.
 
+### laravel_access_control_allow_credentials
+
+Whether or not allow credentials for CORS, Access-Control-Allow-Credentials.
+
 ### laravel_access_control_allow_origin
 
-Enable CORS, and specify Access-Control-Allow-Origin.
+Enable CORS, and specify allowded origin, Access-Control-Allow-Origin.
```
Comment (tjoelsson) [inline README.md]: allowed

File: README.md [hunk 2]
```diff
@@ -57,9 +57,25 @@ Retention of the logrotate.
 
 Default is 30.
 
+### laravel_access_control_allow_credentials
+
+Whether or not allow credentials for CORS, Access-Control-Allow-Credentials.
```
Comment (tjoelsson) [inline README.md]: Whether or not to allow...

File: templates/nginx.conf.j2 [hunk 3]
```diff
@@ -36,28 +37,20 @@ server {
...
-        if ($request_method = 'OPTIONS') {
-            add_header 'Access-Control-Allow-Origin' '{{ laravel_access_control_allow_origin }}';
-            add_header 'Access-Control-Allow-Methods' 'GET,POST,OPTIONS';
-            # Custom headers and headers various browsers *should* be OK with but aren't
-            add_header 'Access-Control-Allow-Headers' 'DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type';
-            # Tell client that this pre-flight info is valid for 20 days
-            add_header 'Access-Control-Max-Age' 1728000;
-            add_header 'Content-Type' 'text/plain charset=UTF-8';
-            add_header 'Content-Length' 0;
-            return 204;
```
Comment (tjoelsson) [inline templates/nginx.conf.j2:49]: Will OPTIONS requests still return 204?

**Review by tjoelsson — CHANGES_REQUESTED** [review]

File: templates/nginx.conf.j2 (hunk 3 above)
Comment (mark-juwai) [reply templates/nginx.conf.j2:49]: No, it doesn't has to be 204, 200 also work.

**Review by tjoelsson — APPROVED** [review]

File: tests/group_vars/all.yml [hunk 4]
```diff
@@ -28,7 +28,11 @@ dot_env:
     - { key: 'pusher_key', value: 'test' }
     - { key: 'pusher_secret', value: 'test' }
 
+laravel_access_control_allow_credentials: false
+laravel_access_control_allow_headers: 'DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type'
+laravel_access_control_allow_methods: 'GET,POST,OPTIONS'
```
Comment (agirivera) [inline tests/group_vars/all.yml]: ? Arrange the headers alphabetically so it will be easier to look for something in the future.

**Review by agirivera — COMMENTED** [review]
I have a non-blocking comment. Also not sure which vagrant box I need to provision to test.

**Comment by mark-juwai** [comment]
>I have a non-blocking comment. Also not sure which vagrant box I need to provision to test.

Running `vagrant up` under folder `tests` will create the vagrant box. Sorry for the unclear description, updated.

**Comment by mark-juwai** [comment]
@agirivera Comment addressed, kindly have a look.

**Review by agirivera — APPROVED** [review]
All good :+1: Saw the relevant sections on the config file.

**Comment by mark-juwai** [comment]
Still good? @tjoelsson

**Review by tjoelsson — APPROVED** [review]

**Pull request merged**